### PR TITLE
Remove MiscDiagnostics diagnosis of invalid InOutExprs

### PR DESCRIPTION
**Pull request by slavapestov**
There was a diagnostic that was only emitted once in the entire test suite. Rework things so that the constraint solver handles this case instead.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please smoke test macOS

File: lib/Sema/CSSimplify.cpp [hunk 1]
```diff
@@ -5605,9 +5597,26 @@ ConstraintSystem::simplifyApplicableFnConstraint(
...
-  // If the types are obviously equivalent, we're done.
-  if (type1.getPointer() == desugar2)
-    return SolutionKind::Solved;
+  SmallVector<LocatorPathElt, 2> parts;
+  Expr *anchor = locator.getLocatorParts(parts);
+  bool isOperator = (isa<PrefixUnaryExpr>(anchor) ||
+                     isa<PostfixUnaryExpr>(anchor) ||
+                     isa<BinaryExpr>(anchor));
+
+  auto hasInOut = [&]() {
```
Comment (xedin) [inline lib/Sema/CSSimplify.cpp]: Nit: This method could be replaced with `llvm::any_of(func1->getParams(), [](const AnyFunctionType::Param &param) { return param.isInOut(); });

**Review by xedin — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**